dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -78,10 +78,14 @@ def authenticate
     extracted_client_id = decoded_jwt_no_auth[:aud]
     extracted_issuer_id = decoded_jwt_no_auth[:iss]
 
-    integration = LtiIntegration.find_by({client_id: extracted_client_id, issuer: extracted_issuer_id})
-    if integration.nil?
-      return log_unauthorized('LTI integration not found', {client_id: extracted_client_id, issuer: extracted_issuer_id})
-    end
+    # set cache key
+    integration_cache_key = "#{extracted_issuer_id}/#{extracted_client_id}"
+
+    integration = read_cache(integration_cache_key) || LtiIntegration.find_by({client_id: extracted_client_id, issuer: extracted_issuer_id})
+    return log_unauthorized('LTI integration not found', {client_id: extracted_client_id, issuer: extracted_issuer_id}) unless integration
+    # Cache integration for fast retrieval on subsequent LTI launches. Set
+    # expires_in to 1 week
+    write_cache(integration_cache_key, integration, 1.week)
```

Comment:
Nice, this is pretty cool that you can store the integration object this easily. Was afraid we'd have to do some weird JSON stringification/parsing

Reply:
I'm doing that down in the `write_function`, stringifying on the way in, parsing on the way out. Had to add some logic to account for a nil cached value.